integrations/sonarqube/integration.py

```diff
@@ -0,0 +1,185 @@
+            filter_instance = SonarQubeComponentSearchFilter(**filter)
+            value["filter"] = filter_instance.generate_search_filters()
+        if s := value.pop("s", None):
+            value["s"] = s
+
+        return value
+
+
+class SonarQubeIssueAPIQueryParams(BaseModel):
+    assigned: Literal["yes", "no", "true", "false"] | None = Field(
+        description="To retrieve assigned or unassigned issues"
+    )
+    assignees: list[str] | None = Field(description="List of assignees")
+    clean_code_attribute_categories: (
+        list[
+            Literal[
+                "ADAPTABLE",
+                "CONSISTENT",
+                "INTENTIONAL",
+                "RESPONSIBLE",
+            ]
+        ]
+        | None
+    ) = Field(
+        alias="cleanCodeAttributeCategories",
+        description="List of clean code attribute categories",
+    )
+    code_variants: list[str] | None = Field(
+        alias="codeVariants", description="List of code variants"
+    )
+    created_before: str | None = Field(
+        alias="createdBefore",
+        description="To retrieve issues created before the given date",
+    )
+    created_after: str | None = Field(
+        alias="createdAfter",
+        description="To retrieve issues created after the given date",
+    )
+    cwe: list[str] | None = Field(description="List of CWE identifiers")
+    impact_severities: list[Literal["HIGH", "LOW", "MEDIUM"]] | None = Field(
+        alias="impactSeverities", description="List of impact severities"
+    )
+    impact_software_qualities: (
+        list[Literal["MAINTAINABILITY", "RELIABILITY", "SECURITY"]] | None
+    ) = Field(
+        alias="impactSoftwareQualities", description="List of impact software qualities"
+    )
+    statuses: (
+        list[Literal["OPEN", "CONFIRMED", "FALSE_POSITIVE", "ACCEPTED", "FIXED"]] | None
+    ) = Field(description="List of statuses")
+    languages: list[str] | None = Field(description="List of languages")
+    owasp_asvs_level: Literal["1", "2", "3"] | None = Field(
+        alias="owaspAsvsLevel", description="OWASP ASVS level"
+    )
+    resolved: Literal["yes", "no", "true", "false"] | None = Field(
+        description="To retrieve resolved or unresolved issues"
+    )
+    rules: list[str] | None = Field(description="List of coding rule keys")
+    scopes: list[Literal["MAIN", "TESTS"]] | None = Field(description="List of scopes")
+    sonarsource_security: list[str] | None = Field(
+        alias="sonarsourceSecurity",
+        description="List of SonarSource security categories",
+    )
+    tags: list[str] | None = Field(description="List of tags")
+
+    def generate_request_params(self) -> dict[str, Any]:
+        value = self.dict(exclude_none=True, by_alias=True)
+
+        for key in list(value.keys()):
+            val = value[key]
+            value[key] = ",".join(val) if isinstance(val, list) else val
+        return value
+
+
+class CustomSelector(Selector):
```

Comment:
make you classes inherit the same class that implement `api_query_params` with default return of `{}` and then just cast the selector to this type if needed